File: cpp/Runner.cpp
#include <iostream>
#include "Child.cpp"

using namespace std;

int main() {
    Child obj;

    vector<string> csvData = obj.readFileData();
    vector<vector<string>> extractedData(csvData.size(), vector<string>(2));

    for (size_t i = 0; i < csvData.size(); i++) {
        string otp = obj.extractOtp(csvData[i]);
        extractedData[i][0] = csvData[i];
        extractedData[i][1] = otp;
    }

    obj.writeFile(extractedData);

    return 0;
}

File: python/Runner.py
from Child import Child

def main():
    obj = Child()
    csv_data = obj.read_file_data()
    extracted_data = []

    for msg in csv_data:
        otp = obj.extract_otp(msg)
        extracted_data.append([msg, otp])

    obj.write_file(extracted_data)

if __name__ == "__main__":
    main()
